detection_rules/config.py

```diff
@@ -100,7 +101,7 @@ def get_test_names(self, formatted: bool = False) -> (List[str], List[str]):
 
         combined_tests = sorted(set(defined_tests + self.tests_by_patterns(*patterns)))
 
-        if self.unit_tests.test_only is not None:
+        if self.unit_tests.test_only:
```

Comment:
can you revert this (the assertion error above was the fix - this is still needed)

```suggestion
        if self.unit_tests.test_only is not None:
```